webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceRepository.scala

```diff
@@ -29,9 +30,11 @@ class DataSourceRepository @Inject()(
     webKnossosServer.reportDataSource(dataSource)
   }
 
-  def updateDataSources(dataSources: List[InboxDataSource]): Unit = {
-    removeAll
-    dataSources.foreach(dataSource => insert(dataSource.id.name, dataSource))
-    webKnossosServer.reportDataSources(dataSources)
-  }
+  def updateDataSources(dataSources: List[InboxDataSource]): Fox[Unit] =
+    for {
+      _ <- Fox.successful(())
```

Comment:
?

Reply:
For comprehensions in scala come in different flavours, selected by the type of the first line. Here we need a Fox-flavoured for-comprehension because of its last line, so the first line has to be a Fox-unpacking too